prereise/gather/demanddata/transportation_electrification/data_process.py

```diff
@@ -0,0 +1,132 @@
+import pandas as pd
+
+#Download CSV of NHTS 2017 from https://nhts.ornl.gov/
+
+def data_filtering(census_division, filepath: str = "trippub.csv"):
+    """Filter raw NHTS data to be used in mileage.py
+    
+    :param str filepath: filepath for the "trippub.csv" file in downloaded NHTS data
+    :return: (*pd.DataFrame*) -- filtered and sorted trip data
+    """
+
+    nhts_raw = pd.read_csv(filepath)
+    
+    #filter to be only vehicle trips (TRPTRANS values 1-6)
+    nhts = nhts_raw[nhts_raw.TRPTRANS.isin(range(1,7))]
+    
+    #filter out repeated trips (VMT = -1)
+    nhts = nhts[nhts.TRPMILES != -1]
+    
+    #get correct census division
+    nhts = nhts[nhts.CENSUS_D == census_division]
+
+    #dataframe for use built here
+    #columns taken directly from original csv:
+    sorted_data = pd.DataFrame({"Household": nhts["HOUSEID"], 
+                   "Vehicle ID": nhts["VEHID"], 
+                   "Person ID": nhts["PERSONID"],
+                   "Scaling Factor Applied": nhts["WTTRDFIN"],
+                   "Trip Number": nhts["TDTRPNUM"],
+                   "Date": nhts["TDAYDATE"], 
+                   "Day of Week": nhts["TRAVDAY"],              
+                   "If Weekend": nhts["TDWKND"],                
+                   "Trip start time (HHMM)": nhts["STRTTIME"],
+                   "Trip end time (HHMM)": nhts["ENDTIME"],
+                   "Travel Minutes": nhts["TRVLCMIN"],
+                   "Dwell time": nhts["DWELTIME"],
+                   "Miles traveled": nhts["VMT_MILE"],          
+                   "Vehicle miles travelled": nhts["TRPMILES"], 
+                   "why from": nhts["WHYFROM"],
+                   "why to": nhts["WHYTO"],
+                   "Vehicle type": nhts["TRPTRANS"],            
+                   "Household vehicle count": nhts["HHVEHCNT"],
+                   "Household size": nhts["HHSIZE"]
+                   })
```

Comment:
Since `nhts` is already a DataFrame, I think we can simplify this using the `rename` method. So instead of:
```python
sorted_data = pd.DataFrame({"oldname1": nhts["newname1"], "oldname2": nhts["newname2"], ...})
```
we could do something like:
```python
column_translations = {"oldname1": "newname1", "oldname2": "newname2", ...}
sorted_data = nhts[column_translations.keys()].rename(column_translations, axis="columns")
```

Reply:
Or we can name overwrite column names using `header=0` and the `names` parameter in `read_csv`.

Reply:
I think if we use the `names` parameter, we'll need to pass a list of names that cover all of the columns, even though we're only interested in a subset of them.

Reply:
And using `usecols` together with `names`? The nice thing with `usecols` is that it can take à `range` or integers.

Reply:
I decided to go with Daniel's first suggestion since not all the columns in the original data set have an alternative name they could map to if we use `names`. But if I overlooked how to use it, let me know and I'll try to implement it!